test/integration/client-side-operations-timeout/client_side_operations_timeout.unit.test.ts

```diff
@@ -1,51 +1,257 @@
+
+      expect(createCalls).to.have.length.greaterThanOrEqual(1);
+      expect(Timeout.expires).to.not.have.been.calledWith(999999);
+    });
+
+    it('If timeoutMS is set for an operation, the remaining timeoutMS value should apply to connection checkout after a server has been selected.', async function () {
+      client = this.configuration.newClient({ timeoutMS: 1000 });
+      // Spy on connection checkout and pull options argument
+      const checkoutSpy = sinon.spy(ConnectionPool.prototype, 'checkOut');
+      const expiresSpy = sinon.spy(Timeout, 'expires');
+
+      await client.db('db').collection('collection').insertOne({ x: 1 });
+
+      expect(checkoutSpy).to.have.been.calledOnce;
+      const timeoutContext = checkoutSpy.lastCall.args[0].timeoutContext;
+      expect(timeoutContext).to.exist;
+      // Check that we passed through the timeout
+      //  @ts-expect-error accessing private properties
+      expect(timeoutContext._serverSelectionTimeout).to.be.instanceOf(Timeout);
+      //  @ts-expect-error accessing private properties
+      expect(timeoutContext._serverSelectionTimeout).to.equal(
+        //  @ts-expect-error accessing private properties
+        timeoutContext._connectionCheckoutTimeout
+      );
+
+      // Check that no more Timeouts are constructed after we enter checkout
+      expect(!expiresSpy.calledAfter(checkoutSpy));
+    });
+
+    it('If timeoutMS is not set for an operation, waitQueueTimeoutMS should apply to connection checkout after a server has been selected.', async function () {
+      client = this.configuration.newClient({ waitQueueTimeoutMS: 123456 });
+
+      const checkoutSpy = sinon.spy(ConnectionPool.prototype, 'checkOut');
+      const selectServerSpy = sinon.spy(Topology.prototype, 'selectServer');
+      const expiresSpy = sinon.spy(Timeout, 'expires');
+
+      await client.db('db').collection('collection').insertOne({ x: 1 });
+      expect(checkoutSpy).to.have.been.calledAfter(selectServerSpy);
+
+      expect(expiresSpy).to.have.been.calledWith(123456);
+    });
+
+    /* eslint-disable @typescript-eslint/no-empty-function */
+    context.skip(
+      'If a new connection is required to execute an operation, min(remaining computedServerSelectionTimeout, connectTimeoutMS) should apply to socket establishment.',
+      () => {}
+    ).skipReason =
+      'TODO(DRIVERS-2347): Requires this ticket to be implemented before we can assert on connection CSOT behaviour';
+
+    context(
+      'For drivers that have control over OCSP behavior, min(remaining computedServerSelectionTimeout, 5 seconds) should apply to HTTP requests against OCSP responders.',
+      () => {}
+    );
+  });
+
+  context.skip('Socket timeouts', function () {
+    context(
+      'If timeoutMS is unset, operations fail after two non-consecutive socket timeouts.',
+      () => {}
+    );
+  }).skipReason =
+    'TODO(NODE-5682): Add CSOT support for socket read/write at the connection layer for CRUD APIs';
```

Comment:
NODE-5682 is closed but this test wasn't implemented

Reply:
Was this supposed to be a spec test? The file isn't spec, so just curious how the placeholder ended up here to begin with

Reply:
Interesting, we should probably note that this is a spec test in some way

Reply:
Addressed in https://github.com/mongodb/node-mongodb-native/pull/4315/files